### PR TITLE
Use static AArch64 decoder tables.

### LINKED ISSUE #630: Static variables in ARM instruction decoder cause crashes
There are static variables in the ARM instruction decoder. 

https://github.com/dyninst/dyninst/blob/29535f22c1a17581107140da3910ccf148b36372/instructionAPI/src/InstructionDecoder-aarch64.C#L96

They should be converted into `std::call_once`.

### PR DESCRIPTION
**Pull request by blue42u**
Fixes #630, reduces compile time (enough to notice), ~7.8MiB smaller binary, ~10MiB less heap usage. Performance effects not well known, but for HPCStruct ~1.2s is saved on a smaller input.

The previous implementation used a lazy initialization scheme to generate the various tables used for arm64 decoding. This PR replaces all that with simple `const` tables and switch statements, which the compiler will usually reduce to a blob in `.data` (or `.text`).

For the small(-ish) lookup tables for the state machine in `main_decoder_table`, a linear scan is employed. Any performance drop is likely caused by this in some way.

Apologies for the unreadable diffs, the only change in `aarch_opcode_tables.C` that isn't noted somewhere else is the removal of duplicate keys in `sysRegMap` (switch wouldn't compile otherwise). I used the last one of every duplicate, hopefully that reflects what the `std::map` would have done.

### CODE REVIEW

File: instructionAPI/src/InstructionDecoder-aarch64.C [hunk 1]
```diff
@@ -29,112 +29,68 @@
...
     namespace InstructionAPI {
         typedef void (InstructionDecoder_aarch64::*operandFactory)();
 
-        typedef std::vector<operandFactory> operandSpec;
-        typedef std::vector<aarch64_insn_entry> aarch64_insn_table;
-        typedef std::map<unsigned int, aarch64_mask_entry> aarch64_decoder_table;
-        typedef std::map<unsigned int, unsigned int> branchMap;
+        typedef aarch64_insn_entry aarch64_insn_table[];
+        typedef aarch64_mask_entry aarch64_decoder_table[];
         typedef uint32_t Bits_t;
```
[the remark this answers is not in the document]
Comment (blue42u) [reply instructionAPI/src/InstructionDecoder-aarch64.C]: Purged in the last few commits.

**Comment by blue42u** [comment]
Ready for another review, if its worth doing before the Capstone stuff is merged.

**Comment by hainest** [comment]
https://bottle.cs.wisc.edu/branch/PR633

Testing for this PR is a bit all of the place. I'm not sure if it's the test suite being it's usual buggy self or if there is a real problem. I will re-run.

**Comment by hainest** [comment]
It's passing on ARM and PPC, and the regressions on x86 are spurious.

**Pull request merged**